### DISCUSSION TITLE
Str::swap()

### DISCUSSION THREAD

**Discussion by khalyomede** [laravel/framework#38884] [Ideas]
Hello everyone,

Recently I needed to inject a bunch of data in a string, and I really did not liked the way it looked because of all the string interpolation that made the string not easily readable.

So I thought I would bring the same "experience" we have when using `trans()` by replacing placeholders:

```php
namespace App\Helpers;

use Illuminate\Support\Str as LaravelStr;

final class Str
{
    /**
     * @param array<string, string> $replace
     */
    public static function swap(string $line, array $replace): string
    {
        $newLine = $line;

        foreach ($replace as $key => $value) {
            $newLine = str_replace(
                [':' . $key, ':' . LaravelStr::upper($key), ':' . LaravelStr::ucfirst($key)],
                [$value, LaravelStr::upper($value), LaravelStr::ucfirst($value)],
                $newLine
            );
        }

        return $newLine;
    }
}
```

All credits to this method: https://github.com/laravel/framework/blob/8.x/src/Illuminate/Translation/Translator.php#L206-L230

Now I can finally make my strings look nicer:

```php
final class MemberAddedToShoppingList extends Notification
{
  public function toDatabase(): array
  {
    return [
      "message" => Str::swap(":author added you on the list :list.", [
        "author" => $this->author->name,
        "list" => $this->list->name,
      ]);
    ];
  }
}
```

What you guys think? Can this be useful for you? I'm eager to make a PR if you guys are interested 🤓

**Comment by devfrey** [comment]
Why though? You can already do exactly this with `__()` and `trans()`, without needing translation files.

**Reply by khalyomede** [reply]
Does not feels legit to use translations helpers for non i18n related string manipulations for me. What you suggested works fine although.
